notebooks/page view functional network/page_view_approach_economic_recovery.ipynb

```diff
@@ -0,0 +1,1507 @@
+    "        docType == 'decision' or docType == 'oral_statement' or docType == 'written_statement' or\n",
+    "        docType == 'authored_article' or docType == 'correspondence' or docType == 'speech' or \n",
+    "        docType == 'government_response' or docType == 'case_study'): \n",
+    "        document_type_dict[docType] = 'news and communication'\n",
+    "    \n",
+    "    elif (docType == 'completed_transaction' or docType == 'local_transaction' or docType == 'form' or \n",
+    "          docType == 'calculator' or docType == 'smart_answer' or docType == 'simple_smart_answer' or \n",
+    "          docType == 'place' or docType == 'licence' or docType == 'step_by_step_nav' or \n",
+    "          docType == 'transaction' or docType == 'answer' or docType == 'guide'):\n",
+    "          document_type_dict[docType] = 'services'\n",
+    "    \n",
+    "    elif (docType == 'regulation' or docType == 'detailed_guide' or docType == 'manual' or \n",
+    "          docType == 'manual_section' or docType == 'guidance' or docType == 'map' or \n",
+    "          docType == 'calendar' or docType == 'statutory_guidance' or docType == 'notice' or \n",
+    "          docType == 'international_treaty' or docType == 'travel_advice' or docType == 'promotional' or \n",
+    "          docType == 'international_development_fund' or docType == 'countryside_stewardship_grant' or docType == 'esi_fund' or \n",
+    "          docType == 'business_finance_support_scheme' or docType == 'statutory_instrument' or docType == 'hmrc_manual' or \n",
+    "          docType == 'standard'):\n",
+    "          document_type_dict[docType] = 'guidance_and_regulation'\n",
+    "\n",
+    "    elif (docType == 'impact_assessment' or docType == 'policy_paper' or docType == 'open_consultation' or \n",
+    "          docType == 'closed_consultation' or docType == 'consultation_outcome'):\n",
+    "          document_type_dict[docType] = 'policy_and_engagement'    \n",
+    "\n",
+    "    elif (docType == 'dfid_research_output' or docType == 'independent_report' or docType == 'research' or \n",
+    "          docType == 'statistics' or docType == 'national_statistics' or docType == 'statistics_announcement' or \n",
+    "          docType == 'national_statistics_announcement' or docType == 'official_statistics_announcement' or \n",
+    "          docType == 'statistical_data_set' or docType == 'official_statistics'):\n",
+    "          document_type_dict[docType] = 'research_and_statistics'    \n",
+    "\n",
+    "    elif (docType == 'transparency' or docType == 'corporate_report' or docType == 'foi_release' or \n",
+    "          docType == 'aaib_report' or docType == 'raib_report' or docType == 'maib_report'):\n",
+    "          document_type_dict[docType] = 'transparency'    \n",
+    "    \n",
+    "    else: \n",
+    "        document_type_dict[docType] = 'other' \n",
```

Comment:
That's a lot of `or`s! Unless you really enjoy writing the word `or`, you don't need to do this. You could do something like this instead:

```
# define a set for news and communication doc types
news_and_comms_doctypes = {
    'medical_safety_alert', 'drug_safety_update', 'news_article',
    'news_story', 'press_release', etc...
}

# define a set for service doc types
service_doctypes = {
    ...
}

# and so on for the others
```

Then do this:
```
for docType, docSupertype in document_type_dict.items():
    if docType in news_and_comms_doctypes: 
        document_type_dict[docType] = 'news and communication'
    
    elif docType in service_doctypes:
        document_type_dict[docType] = 'services'

    # and so on...
```

Reply:
Aha, much nicer! Thank you Jake. Resolved in commit 82c2170